lib/kube/proxy/forwarder.go

```diff
@@ -1575,39 +1547,54 @@ func (f *Forwarder) newClusterSessionDirect(ctx authContext, endpoints []kubeClu
 	}
 
 	f.log.WithField("kube_service.endpoints", endpoints).Debugf("Kubernetes session for %v forwarded to remote kubernetes_service instance.", ctx)
-	sess := &clusterSession{
+	return &clusterSession{
 		parent:               f,
 		authContext:          ctx,
 		kubeClusterEndpoints: endpoints,
 		tlsConfig:            tlsConfig,
 		// This session talks to a kubernetes_service, which should handle
 		// audit logging. Avoid duplicate logging.
 		noAuditEvents: true,
+	}, nil
+}
+
+func (f *Forwarder) setSessionForwarder(sess *clusterSession) error {
+	var err error
+	transport := f.newTransport(sess.Dial, sess.tlsConfig)
+
+	if sess.upgradeToHTTP2 {
+		transport = utilnet.SetTransportDefaults(transport)
+	} else {
+		sess.tlsConfig.NextProtos = []string{}
 	}
```

Comment:
Nit: Explain with a comment why we sometimes want to upgrade to HTTP2 and sometimes we don't.